pkg/template/helper.go

```diff
@@ -0,0 +1,48 @@
+package template
+
+import "reflect"
+
+// VisitFields visits all fields in the object and call the visitor function on
+// them. In case the object has no fields (like string) then it calls the
+// visitor function on it.
+// The visitor function can manipulate the fields values.
+func VisitFields(obj interface{}, visitor func(string) string) {
+	v := reflect.ValueOf(obj).Elem()
+
+	visit := func(val reflect.Value) {
+		switch val.Kind() {
+		case reflect.String:
+			if val.CanSet() {
+				val.SetString(visitor(val.String()))
+			}
+		case reflect.Ptr, reflect.Interface:
+			if val.Elem().Kind() == reflect.Struct {
+				VisitFields(val.Interface(), visitor)
+			}
+		case reflect.Struct:
```

Comment:
You can have an array of arrays, and an array of maps, so this code doesn't look right to me.